Core/MIPS/IR/IRPassSimplify.cpp

```diff
@@ -0,0 +1,199 @@
+#include "Core/MIPS/IR/IRPassSimplify.h"
+#include "Core/MIPS/IR/IRRegCache.h"
+
+void SimplifyInPlace(IRInst *inst, int count, const u32 *constPool) {
+	for (int i = 0; i < count; i++) {
+		switch (inst[i].op) {
+		case IROp::AddConst:
```

Comment:
Right.  Although, I suppose some in-place optimizations (like combining, by nop'ing) could be done, it probably belongs in other passes.

-[Unknown]